backend/src/modules/task/services/task.service.ts

```diff
@@ -56,31 +60,52 @@ export class TaskService {
     };
   }
 
-  async getAllTasks() {
+  async getAllTasks(filterDto: TaskFilterDto) {
+    const where = this.buildTaskFilter(filterDto);
+  
     const tasks = await this.prisma.task.findMany({
-      where: {
-        deletedAt: null,
-      },
+      where,
+      orderBy: { taskOrder: "asc" },
     });
+  
+    return { message: "Filtered tasks retrieved successfully", tasks };
+  }
+  
 
-    return {
-      message: 'All tasks retrieved successfully',
-      tasks,
-    };
+  private buildTaskFilter(filterDto: TaskFilterDto) {
+    const { status, createdBy, assignedTo } = filterDto;
+    const where: any = { deletedAt: null };
+
+    if (status && status !== 'ALL' && Object.values(Status).includes(status as Status)) {
```

Comment:
like to asked you before, and actually commented as well, please do not send ALL to BE